src/main/resources/api/service_openapi.yaml

```diff
@@ -247,9 +247,7 @@ paths:
       responses:
         '204':
           description: Role granted successfully
-        '403':
-          $ref: '#/components/responses/PermissionDenied'
-        '404':
+        '401':
```

Comment:
401 is not the right response. UNAUTHORIZED means that your credentials were invalid - not that you do not have permission. See https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401. It would be misleading to return.

We can confirm with others, but my understanding is that for MC Terra APIs we do not need to hide the difference between NOT FOUND and FORBIDDEN. I know that in secure APIs, the best practice is to say NOT FOUND uniformly to not leak existence info.

Reply:
Sam does not hide the difference, FWIW

Reply:
Ah, I was under the impression that we were trying to avoid leaking existence info. Which parts of Sam don't hide the difference? querying "can I do action X on resource Y" returns `false` for things that don't exist, or things where it's actually false

Reply:
I was looking at:
    GET /api/resources/v1/{resourceTypeName}/{resourceId}/policies
but there are a fair number that return both 403 and 404.

Reply:
I think for CWB, our approach has been that WSM doesn't need to hide any error codes, and we should handle leak-avoidance at a higher level. I'm not sure if that approach holds true for your non-CWB uses ... is WSM still a low-level service in that architecture?

Reply:
I checked with Sarah and Denis. There is no compliance reason for us to be hiding the state.

I think hiding it is a negative, since it makes it harder for the caller to debug what has gone wrong.